kolibri/utils/server.py

```diff
@@ -285,22 +296,24 @@ def get_status():  # noqa: max-complexity=16
     :raises: NotRunning
     """
 
-    # There is no PID file (created by server daemon)
-    if not os.path.isfile(PID_FILE):
-        # Is there a startup lock?
-        if os.path.isfile(STARTUP_LOCK):
-            try:
-                pid, port = _read_pid_file(STARTUP_LOCK)
-                # Does the PID in there still exist?
-                if pid_exists(pid):
-                    raise NotRunning(STATUS_STARTING_UP)
-                # It's dead so assuming the startup went badly
-                else:
-                    raise NotRunning(STATUS_FAILED_TO_START)
-            # Couldn't parse to int or empty PID file
-            except (TypeError, ValueError):
-                raise NotRunning(STATUS_STOPPED)
-        raise NotRunning(STATUS_STOPPED)  # Stopped
+    # Is there a startup lock?
+    if os.path.isfile(STARTUP_LOCK):
```

Comment:
From your comment I am not sure if you are just explaining it or requesting something more to be done.

Reply:
I am not explaining anything else to be done, just that since you have recently looked at the code, there is a bit of background to it. I did not put a giant "startup lock" around everything because there was a concern that we would get in to a situation where Kolibri's own startup lock mechanism was blocking it from starting up.

This is just background, otherwise I would have mentioned it in the actual review.

Reply:
Below we check whether the PID exists and raise an exception if it doesn't. Perhaps a safer action would be to log a warning, delete the startup lock, and continue with startup?